packages/protocol/contracts/bridge/TokenVault.sol

```diff
@@ -388,7 +389,10 @@ contract TokenVault is EssentialContract {
         bridgedToken = Create2Upgradeable.deploy(
             0, // amount of Ether to send
             keccak256(
-                abi.encodePacked(canonicalToken.chainId, canonicalToken.addr)
+                bytes.concat(
+                    bytes32(canonicalToken.chainId),
+                    bytes32(uint256(uint160(canonicalToken.addr)))
+                )
```

Comment:
This kind of conversion: `bytes32(integer)` results in empty bytes (and thus strings when converted). Every type of `bytes32(integer)` is affected (e.g.: in LibProving.sol 
```
buffer = bytes.concat(
                    buffer,
                    bytes32(uint256(uint160(evidence.prover))),
                    bytes32(uint256(evidence.meta.id)),
                    bytes32(evidence.meta.l1Height),
                    evidence.meta.l1Hash,
                    evidence.meta.txListHash
                );
```
 )

(BTW this is why the hardhat tests were failing @dantaik when you tested here https://github.com/taikoxyz/taiko-mono/issues/13290 , because silently - as foundry is also written in Solidity - it appended the 'empty bytes' to the others, so you observed no failed tests.)

Best way to test this out is, for example within a foundry environment, find attached this code: (both hashes are the same).
```
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.18;

import "forge-std/Test.sol";
import "forge-std/console.sol";

contract FooBar {
   
    function _deployBridgedToken() public {
        address dummyCanonicalTokenAddress = address(this);
        uint256 chainId = 1;
        uint256 chainId2 = 2;

        bytes32 keccakedConcatenation =
            keccak256(
                bytes.concat(
                    bytes32(chainId),
                    bytes32(uint256(uint160(dummyCanonicalTokenAddress)))
                )
        );

        console.log("Output with chain ID 1");
        console.log(string(abi.encodePacked(keccakedConcatenation)));

        bytes32 keccakedConcatenation2 =
            keccak256(
                bytes.concat(
                    bytes32(chainId2),
                    bytes32(uint256(uint160(dummyCanonicalTokenAddress)))
                )
        );

        console.log("Output with chain ID 2:");
        console.log(string(abi.encodePacked(keccakedConcatenation)));
    }
}

contract EmptyBytesTest is Test {
    FooBar foobar;

    function setUp() public {
        foobar = new FooBar();
    }

    function testBytes() external {
        foobar._deployBridgedToken();
    }
    
}
```